### DISCUSSION TITLE
Jedis 4+ dropped support for multi inside pipeline

### DISCUSSION THREAD

**Discussion by pgilad** [xetorthio/jedis#2993] [Q&A]
On version 4+ of Jedis, transaction (multi + exec) inside a pipelined transaction was dropped:
<img width="855" alt="image" src="https://user-images.githubusercontent.com/4533329/168172541-8ba50475-ddef-4e30-8552-e34f93f81628.png">

Why did you apply this change, and what is the alternative way today on version v4+ to indeed run a transaction that's pipelined? (MULTI + EXEC inside pipelined) ?

**Comment by sazzad16** [comment]
The `Transaction` class is backed by pipelining.

**Reply by pgilad** [reply]
It doesn't look like `Transaction` is pipelined. For example `multi()` calls

```
public final void multi() {
    connection.sendCommand(MULTI);
    processMultiResponse();
    inMulti = true;
  }
```

Which actually sends the command to server.
And if we look at `appendCommand` in `TransactionBase` then it also sends the commands to server:

```
protected final <T> Response<T> appendCommand(CommandObject<T> commandObject) {
    connection.sendCommand(commandObject.getArguments());
    processAppendStatus();
    return enqueResponse(commandObject.getBuilder());
  }
```

So it doesn't look backed by pipelining at all. What am I missing?
